worlds/kh1/Rules.py

```diff
@@ -53,487 +81,570 @@ def has_all_magic_lvx(state: CollectionState, player: int, level) -> bool:
-        return has_puppies_all(state, player, 99)
-    if final_rest_door_requirement == "postcards":
-        return state.has("Postcard", player, 10)
-    if final_rest_door_requirement == "superbosses":
-        return (
-                state.has_all({"Olympus Coliseum", "Neverland", "Agrabah", "Hollow Bastion", "Green Trinity", "Phil Cup", "Pegasus Cup", "Hercules Cup", "Entry Pass"}, player)
-                and has_emblems(state, player, keyblades_unlock_chests)
-                and has_all_magic_lvx(state, player, 2)
-                and has_defensive_tools(state, player)
-                and has_x_worlds(state, player, 7, keyblades_unlock_chests)
-            )
+def has_final_rest_door(state: CollectionState, player: int, final_rest_door_requirement: str, final_rest_door_required_lucky_emblems: int):
+    if final_rest_door_requirement == "lucky_emblems":
+        return state.has("Lucky Emblem", player, final_rest_door_required_lucky_emblems)
+    else:
+        return state.has("Final Door Key", player)
 
-def has_defensive_tools(state: CollectionState, player: int) -> bool:
-    return (
+def has_defensive_tools(state: CollectionState, player: int, logic_difficulty: int) -> bool:
+    if logic_difficulty > 14:
+        return True
+    else:
+        return (
             state.has_all_counts({"Progressive Cure": 2, "Leaf Bracer": 1, "Dodge Roll": 1}, player)
             and state.has_any_count({"Second Chance": 1, "MP Rage": 1, "Progressive Aero": 2}, player)
         )
 
+def has_basic_tools(state: CollectionState, player: int) -> bool:
+    return (
+            state.has_all_counts({"Dodge Roll": 1, "Progressive Cure": 1}, player)
+            and state.has_any_count({"Combo Master": 1, "Strike Raid": 1, "Sonic Blade": 1, "Counterattack": 1}, player)
+            and state.has_any_count({"Leaf Bracer": 1, "Second Chance": 1, "Guard": 1}, player)
+            and has_offensive_magic(state, player, 6)
+        )
+
 def can_dumbo_skip(state: CollectionState, player: int) -> bool:
     return (
             state.has("Dumbo", player)
             and state.has_group("Magic", player)
         )
 
-def has_oogie_manor(state: CollectionState, player: int, advanced_logic: bool) -> bool:
+def has_oogie_manor(state: CollectionState, player: int, logic_difficulty: int) -> bool:
     return (
-                state.has("Progressive Fire", player)
-                or (advanced_logic and state.has("High Jump", player, 2))
-                or (advanced_logic and state.has("High Jump", player) and state.has("Progressive Glide", player))
+            state.has("Progressive Fire", player)
+            or (logic_difficulty > 0 and state.has("High Jump", player, 3))
+            or (logic_difficulty > 5 and state.has("High Jump", player, 2) or (state.has_all({"High Jump", "Progressive Glide"}, player)))
+            or (logic_difficulty > 10 and state.has_any({"High Jump", "Progressive Glide"}, player))
```

Comment:
Just confirming since this looks like an easy place to make a mistake: the intent here is for this to only apply if logic_difficulty is set to minimal, right?